console/console-init/ui/src/modules/project/containers/ProjectToolbarContainer/ProjectToolbarContainer.tsx

```diff
@@ -0,0 +1,267 @@
+import { ISearchNameOrNameSpaceAddressSpaceListResponse } from "schema/ResponseTypes";
+import { IProjectFilter } from "modules/project/ProjectPage";
+import { initialiseFilterForProject } from "modules/project/utils";
+import { ProjectToolbar } from "modules/project/components/ProjectToolbar";
+
+export interface IProjectToolbarContainerProps {
+  filter: IProjectFilter;
+  setFilter: (filter: IProjectFilter) => void;
+  totalProjects: number;
+  sortValue?: ISortBy;
+  setSortValue: (value: ISortBy) => void;
+  onDeleteAll: () => void;
+  isDeleteAllDisabled: boolean;
+  onSelectAllProjects: (val: boolean) => void;
+  isAllProjectSelected: boolean;
+}
+
+export const ProjectToolbarContainer: React.FunctionComponent<IProjectToolbarContainerProps> = ({
+  filter,
+  setFilter,
+  totalProjects,
+  sortValue,
+  setSortValue,
+  onDeleteAll,
+  isDeleteAllDisabled,
+  onSelectAllProjects,
+  isAllProjectSelected
+}) => {
+  const client = useApolloClient();
+  // const { dispatch } = useStoreContext();
+  const [nameSelected, setNameSelected] = useState<string>();
+  const [nameInput, setNameInput] = useState<string>();
+  const [namespaceSelected, setNamespaceSelected] = useState<string>();
+  const [namespaceInput, setNamespaceInput] = useState<string>();
+
+  const onClearAllFilters = () => {
+    setFilter(initialiseFilterForProject());
+  };
+
+  const onSelectDeleteAll = async (event: any) => {
+    if (event.target.value === "deleteAll") {
+      await onDeleteAll();
+    }
+  };
+
+  //this function used to clear value of type ahead select input field on filter change
+  const resettInitialState = () => {
+    setNameSelected(undefined);
+    setNameInput(undefined);
+    setNamespaceInput(undefined);
+    setNamespaceSelected(undefined);
+  };
+
+  const onFilterSelect = (value: string) => {
+    setFilter({ ...filter, filterType: value });
+    resettInitialState();
+  };
+
+  const onNameSelect = (e: any, selection: SelectOptionObject) => {
+    setNameSelected(selection.toString());
+    setNameInput(undefined);
+  };
+
+  const onNameClear = () => {
+    setNameSelected(undefined);
+    setNameInput(undefined);
+  };
+
+  const getAddressSpaceForNameORNameSpace = async (
+    value: string,
+    propertyName: string
+  ) => {
+    const response = await client.query<
+      ISearchNameOrNameSpaceAddressSpaceListResponse
+    >({
+      query: RETURN_ALL_ADDRESS_SPACES_FOR_NAME_OR_NAMESPACE(
+        propertyName,
+        value.trim()
+      ),
+      fetchPolicy: FetchPolicy.NETWORK_ONLY
+    });
+    return response;
+  };
+
+  const getOptions = (
+    response: ISearchNameOrNameSpaceAddressSpaceListResponse,
+    propertyname: string
+  ) => {
+    if (
+      response.addressSpaces &&
+      response.addressSpaces.addressSpaces &&
+      response.addressSpaces.addressSpaces.length > 0
+    ) {
+      let obtainedList = response.addressSpaces.addressSpaces.map(
```

Comment:
obtainedList looks like a list of address spaces, can you please name it accordingly?

Reply:
It will be modified once I implement the query